### PR TITLE
New use cases for modification of model and component hierarchy

### PR DESCRIPTION
**Pull request by nickerso**
add initial draft of new use cases for the modification of an existing model and component hierarchy. Addressing issue #33

### CODE REVIEW

File: docs/usecases.rst [hunk 1]
```diff
@@ -20,6 +20,17 @@ Use-cases for libCellML
...
       #. one component encapsulating a single child which in turn encapsulates a single child
       #. an invalid cyclical encapsulation hierarchy
+      
+   #. manipulation of a model with multi-level component encapsulation hierarchy
+   
+      a. remove a component
+      #. change the name of a top-level component
+      #. change the name of an encapsulated child component
+      #. replace one component with a new component
+      #. take a component
```
Comment (rchristie) [inline docs/usecases.rst]: 'take' is ambiguous to me, could mean either get a component from a model (model is unchanged) or remove a component from a model.

**Comment by rchristie** [comment]
Looks good to me, with the last minor change.

**Comment by mirams** [comment]
I think these sound sensible tasks for use cases.

File: docs/usecases.rst [hunk 2]
```diff
@@ -20,6 +20,16 @@ Use-cases for libCellML
       a. one component encapsulating two children
       #. one component encapsulating a single child which in turn encapsulates a single child
       #. an invalid cyclical encapsulation hierarchy
+      
+   #. manipulation of a model with multi-level component encapsulation hierarchy
+   
+      a. remove a component
```
Comment (agarny) [inline docs/usecases.rst]: As for changing the name of a component, shouldn't we have both "remove a top-level component" and "remove an encapsulated child component"?

Comment (agarny) [inline docs/usecases.rst]: A part from my comment above, this looks good to me.

Comment (agarny) [inline docs/usecases.rst]: In fact, looking at what has been implemented so far (see #36), it looks like we should make it clear that we also want to test the removal of a component from a model (and not just a component of sorts).

Comment (nickerso) [inline docs/usecases.rst]: good point, I've added those now.

**Pull request merged**